c10/util/BFloat16-math.h

```diff
@@ -10,95 +11,188 @@ C10_CLANG_DIAGNOSTIC_IGNORE("-Wimplicit-float-conversion")
 
 namespace std {
 
-/// Used by vec256<c10::BFloat16>::map
-inline c10::BFloat16 acos(c10::BFloat16 a) {
+template <typename T>
+struct is_reduced_floating_point
+    : std::integral_constant<
+          bool,
+          std::is_same<T, c10::Half>::value ||
+              std::is_same<T, c10::BFloat16>::value> {};
+
+template <
+    typename T,
+    typename std::enable_if_t<is_reduced_floating_point<T>::value, int> = 0>
```

Comment:
nit: use `is_reduced_floating_point_v` here and the rest of code

Reply:
Done, thanks.